pkg/server/sotw/v2/server.go

```diff
@@ -256,7 +304,30 @@ func (s *server) process(stream Stream, reqCh <-chan *discovery.DiscoveryRequest
 						return err
 					}
 				}
-				values.cancellations[typeUrl] = s.cache.CreateWatch(req, values.responses)
+				value := values.responses
+				if s.unordered {
```

Comment:
i think in the older model, cancelWatch was guarenteed to be called before createWatch everytime

Reply:
I think it still happens before CreateWatch at line 300?